compiler/AST/expr.cpp

```diff
@@ -5808,12 +5816,31 @@ GenRet CallExpr::codegenPrimitive() {
 GenRet CallExpr::codegenPrimMove() {
   GenRet ret;
 
+  GenRet specRet;
   if (get(1)->typeInfo() == dtVoid) {
     ret = get(2)->codegen();
 
   // Is the RHS a primop with special case handling?
-  } else if (codegenPrimMoveRhsIsSpecialPrimop() == true) {
-    // Codegen has been handled.  No more activity here.
+  } else if (codegenIsSpecialPrimitive(get(1), get(2), specRet)) {
+    CallExpr* rhsCe = toCallExpr(get(2));
+    if(rhsCe->isPrimitive(PRIM_GET_REAL) || rhsCe->isPrimitive(PRIM_GET_IMAG)) {
+
+      if (gGenInfo->cfile) {
+        std::string stmt = codegenValue(get(1)).c + " = ";
+
+        stmt += specRet.c;
+        stmt += ";\n";
+
+        gGenInfo->cStatements.push_back(stmt);
+      } else {
+#ifdef HAVE_LLVM
+        codegenStoreLLVM(*specRet, get(1));
+#endif
+      }
```

Comment:
2 questions:
1) Why do PRIM_GET_REAL and PRIM_GET_IMAG need special treatment here? Why can't that be handled in codegenIsSpecialPrimitive?
2) Even if PRIM_GET_REAL/IMAG need special logic here, couldn't you use codegenAssign either way, and just adjust the arguments to it?